### PR TITLE
Update to read CRDs from `v1`

### PR DESCRIPTION
**Pull request by howardjohn**
This changes all of our code to read `v1` CRDs, instead of `v1alpha3`/`v1beta1`.

This would break users only if they have 1.21 Istio CRDs with Istio 1.24+, which is a very odd setup and not supported anyways.

The current set of watches now:
```
/admissionregistration.k8s.io/v1/mutatingwebhookconfigurations,null,null
/admissionregistration.k8s.io/v1/validatingwebhookconfigurations,["istio.io/rev="],null
/apiextensions.k8s.io/v1/customresourcedefinitions,null,null
/apps/v1/deployments,null,null
/discovery.k8s.io/v1/endpointslices,null,null
/extensions.istio.io/v1alpha1/wasmplugins,null,null
/gateway.networking.k8s.io/v1beta1/gatewayclasses,null,null
/gateway.networking.k8s.io/v1beta1/gateways,null,null
/gateway.networking.k8s.io/v1beta1/httproutes,null,null
/gateway.networking.k8s.io/v1beta1/referencegrants,null,null
/gateway.networking.k8s.io/v1/grpcroutes,null,null
/networking.istio.io/v1alpha3/envoyfilters,null,null
/networking.istio.io/v1beta1/proxyconfigs,null,null
/networking.istio.io/v1/destinationrules,null,null
/networking.istio.io/v1/gateways,null,null
/networking.istio.io/v1/serviceentries,null,null
/networking.istio.io/v1/sidecars,null,null
/networking.istio.io/v1/virtualservices,null,null
/networking.istio.io/v1/workloadentries,null,null
/networking.istio.io/v1/workloadgroups,null,null
/networking.k8s.io/v1/ingressclasses,null,null
/networking.k8s.io/v1/ingresses,null,null
/security.istio.io/v1/authorizationpolicies,null,null
/security.istio.io/v1/peerauthentications,null,null
/security.istio.io/v1/requestauthentications,null,null
/telemetry.istio.io/v1/telemetries,null,null
/v1/namespaces/istio-system/configmaps,null,["metadata.name=istio"]
/v1/namespaces/istio-system/configmaps,null,["metadata.name=istio-sidecar-injector"]
/v1/namespaces/istio-system/secrets,["istio/multiCluster=true"],null
/v1/namespaces,null,null
/v1/nodes,null,null
/v1/pods,null,null
/v1/secrets,null,["type!=helm.sh/release.v1,type!=kubernetes.io/service-account-token"]
/v1/serviceaccounts,null,null
/v1/services,null,null
```

This PR is split into 3 commits, if it helps review:
* Update codegen 
* Update Go usage
* Update yaml usage

### CODE REVIEW

**Review by MorrisLaw — APPROVED** [review]
Looked through a good portion of this. The change seems sane to me, so I'll approve to make sure it's not blocked on the UX side of things.

**Review by lei-tang — APPROVED** [review]

**Review by craigbox — APPROVED** [review]

File: pilot/pkg/serviceregistry/kube/controller/ambient/ambientindex_test.go [hunk 1]
```diff
@@ -31,8 +31,8 @@ import (
 	"istio.io/api/networking/v1alpha3"
 	auth "istio.io/api/security/v1beta1"
 	"istio.io/api/type/v1beta1"
-	apiv1alpha3 "istio.io/client-go/pkg/apis/networking/v1alpha3"
-	clientsecurityv1beta1 "istio.io/client-go/pkg/apis/security/v1beta1"
+	apiv1alpha3 "istio.io/client-go/pkg/apis/networking/v1"
```
Comment (hzxuzhonghu) [inline pilot/pkg/serviceregistry/kube/controller/ambient/ambientindex_test.go:34]: change it

**Review by hzxuzhonghu — COMMENTED** [review]
when do we change the stored version?

**Comment by howardjohn** [comment]
> when do we change the stored version?

I'll need to look into it more for edge cases but I think we can do it any time. It is not coupled with this change though

**Review by zirain — APPROVED** [review]

**Comment by craigbox** [comment]
tests passing, just one review to go!

**Review by kfaseela — APPROVED** [review]

**Review by stevenctl — APPROVED** [review]
